src/core/token.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <boost/uuid/uuid_generators.hpp>
+#include <string>
+
+class Token {
```

Comment:
This can not be `Token` if it creates a token. Maybe token should be generated in constructor or this class should have a different name